src/blocklyc.js

```diff
@@ -1069,18 +1053,12 @@ var graphStartStop = function (action) {
 var check_com_ports = function () {
     // TODO: We need to evaluate this when using web sockets ('ws') === true
     if (client_use_type !== 'ws') {
-        if (client_url !== undefined) {
-            if (client_version >= minVer) {
-                // Client is >= minimum supported version
-                $.get(client_url + "ports.json", function (data) {
-                    set_port_list(data);
-                }).fail(function () {
-                    set_port_list();
-                });
-            } else {
-                // else keep port list clear (searching...)
+        if (client_domain_name && client_domain_port) {
```

Comment:
These variables are globally set and always have values, so there is no need to verify their existence

Reply:
Let's get the global variables into the globals.js file. Better to herd the globals into one place so we can thin them out with classes.

Reply:
Next PR has more work on this, but yes, they will get there.